website/docs/getting-started.md

```diff
@@ -0,0 +1,153 @@
+
+Desired proprieties are a feature to allow for a smaller memory impact on your application for proprieties that you don't use. An example may be the channel topic, you might not be interested in knowing the topic of a channel, however Discord will always give it to you making your code have to deal with values that you will not use but still consume memory. This is why Discordeno requires you to explicitly set ahead of time the proprieties that you want/use.
+
+You can set these from the `bot.transformers.desiredProprieties` object. This will contain an object for every type that supports this feature, such as `Message` or `Interaction` for example where inside you will find the single proprieties with a boolean value, or an object with more proprieties inside. Discordeno will default everything to false and you can set the values you need to true manually like this:
+
+```ts
+bot.transformers.desiredProprieties.message.id = true
+bot.transformers.desiredProprieties.message.content = true
+bot.transformers.desiredProprieties.message.channelId = true
+```
+
+With the above 3 lines of code we will get the message ID, the channel ID and the message content\* for a specific message, and if we now see the object from both our sendMessage helper method and/or the messageCreate we will find the `id`, the `channelId` and the `content` proprieties to be present in the object.
+
+We could also the proprieties using the Javascript Object Syntax (`{ id: true, content: true, channelId: true }`), but that would require us to specify all others keys in the object with false, and doing something like that would get extremely annoying.
+
+\*: As long as the required privileged intent is enabled.
+
+:::danger[Changing the default for Desired Proprieties]
+THIS IS NOT RECOMMENDED IF YOU WANT TO SHIP YOUR BOT TO PRODUCTION.
+
+While not recommended, in the `createBot` we do allow for a value: `defaultDesiredPropertiesValue`. This if set to true every desired propriety by default, you can still disable some if you need. The reason why this is not recommended and marked as deprecated from the documentation is because while Desired Proprieties DO slow you down during development needing to make sure you aren't using something that you won't have a runtime they have a significant performance impact based on the proprieties you disable on both a CPU side and memory side. Also we are currently working on a solution to the current Typescript issues (see box below) and that may remove the `defaultDesiredPropertiesValue` value all together or move it to a new place.
```

Comment:
```suggestion
While not recommended, in the `createBot` function, we do allow for a value: `defaultDesiredPropertiesValue`. This, if set to true will set every desired property to true by default, you can still disable some if you need. The reason why this is not recommended and marked as deprecated from the documentation is because while Desired Properties DO slow you down during development (needing to make sure you aren't using something that you won't have at runtime), they have a significant performance impact based on the properties you disable on both a CPU side and memory side. Also, we are currently working on a solution to the current TypeScript issues (see box below) and that may remove the `defaultDesiredPropertiesValue` value altogether or move it to a new place.
```